conn.go

```diff
@@ -131,6 +139,8 @@ type connect struct {
 	readTimeout          time.Duration
 	blockBufferSize      uint8
 	maxCompressionBuffer int
+	mutex                sync.Mutex
```

Comment:
do you think it makes sense to name it explicit?
```suggestion
	readerMutex                sync.Mutex
```

Reply:
yeah, good idea

Reply:
done